### PR TITLE
Fix our xz2 crate configuration to enable static linking.

### LINKED ISSUE #172: Unexpected dependency to homebrew xz library from a scie executable
```
> otool -L ./backendai-install-macos-aarch64
./backendai-install-macos-aarch64:
        /opt/homebrew/opt/xz/lib/liblzma.5.dylib (compatibility version 8.0.0, current version 8.9.0)
        /usr/lib/libiconv.2.dylib (compatibility version 7.0.0, current version 7.0.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1336.0.0)
```

I discovered that executing my scie-built executable on one of "fresh" macOS setup fails due to a missing dylib: `/opt/homebrew/opt/xz/lib/liblzma.5.dylib`.

I think it should be statically linked into the scie binary.

### PR DESCRIPTION
**Pull request by jsirois**
Previously liblzma was dynamically linked (to a homebrew version).

Fixes #172

### CODE REVIEW

**Comment by jsirois** [comment]
The linking is now:
```
$ otool -L dist/scie-jump-macos-aarch64
dist/scie-jump-macos-aarch64:
	/usr/lib/libiconv.2.dylib (compatibility version 7.0.0, current version 7.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1336.0.0)
```

**Comment by jsirois** [comment]
The fix is good, but I'm in a weird state where `cargo +nightly fmt` is giving different results on the Linux and Mac runners.

**Review by huonw — APPROVED** [review]
I'm assuming that the `Cargo.toml` change is the only real change, and the code changes are "just" reformatting?

Are you aware that rustup supports pinning to particular nightlies via `nightly-$date`? For instance, `rustup toolchain add nightly-2023-11-20`. https://rust-lang.github.io/rustup/concepts/toolchains.html#toolchain-specification This might allow getting stable formatting that only changes when explicitly desired, or at least pinning to a version that doesn't have the inconsistencies.

**Comment by jsirois** [comment]
> Are you aware that rustup supports pinning to particular nightlies via nightly-$date

Yeah, but I've never had to do this until now - totally unclear why. Even having to use nightly at all is horrible. I'm not sure what is up with the glacial pace of the promotion of rustfmt nightly features compared to rust otherwise.

**Comment by jsirois** [comment]
This is a very crappy situation. Wanting to use stable, and encoding that in rust-toolchain so all devs get things working, but then using nightly for rustfmt. Afaict I'm forced to add docs or an outer script or both to make sure fmt just works as well. Having to use a magic nightly pinned version is going to skate right by a new dev. In fact, it will mess me up too I;m sure when I forget a month from now.

**Comment by jsirois** [comment]
I'm going to sink some time into this rustfmt reproducibility situation before landing.

**Review by kaos — APPROVED** [review]

**Comment by jsirois** [comment]
Alright, much headbanging but I should have looked at issues: https://github.com/rust-lang/rustfmt/issues/5964

Still not sure what to do, but this appears to be a ~temporary problem.

**Comment by jsirois** [comment]
Alright - the rustfmt noise is isolated to some ~commented out CI steps and tracked in https://github.com/a-scie/jump/issues/174.

I'll merge once green since those CI fixes are minimal despite being orthogonal.

**Pull request merged**
